components/track/TrackItem.vue

```diff
@@ -1,28 +1,92 @@
 <script lang="ts" setup>
 import { TrackModel } from "@bcc-code/bmm-sdk-fetch";
 
-defineProps<{
+const props = defineProps<{
   track: TrackModel;
   showThumbnail?: boolean;
 }>();
+
+const trackitemInnerClasses = computed(() => ({
+  "trackitem-inner": true,
+  "has-thumbnail": props.track.meta?.attachedPicture && props.showThumbnail,
+}));
 </script>
 
 <template>
-  <li class="py-3 rounded-xl flex gap-3 items-center">
-    <ProtectedImage
-      v-if="track.meta?.attachedPicture && showThumbnail"
-      :src="track.meta?.attachedPicture"
-      alt=""
-      class="rounded-md w-10 aspect-square bg-slate-100"
-    />
-    <div class="w-1/2">
-      <h4 class="font-semibold">{{ track.meta?.title }}</h4>
-      <span class="text-slate-700" v-if="track.meta?.artist">
-        {{ track.meta?.artist }}
-      </span>
-    </div>
-    <div class="">
-      <span class="text-slate-400">{{ track.subtype }}</span>
+  <li class="trackitem" tabindex="0">
+    <div :class="trackitemInnerClasses">
+      <div v-if="track.meta?.attachedPicture && showThumbnail" class="w-full relative">
+        <ProtectedImage :src="track.meta?.attachedPicture" alt="" class="rounded-md w-full aspect-square bg-slate-100" />
+        <div class="only-show-on-hover absolute inset-0">
+          <div class="absolute inset-0 rounded-md bg-black opacity-50 w-full h-full"></div>
+          <Icon name="play" filled class="absolute inset-0 text-white text-2xl flex justify-center items-center" />
+        </div>
+      </div>
+      <div class="track-title-and-artist">
+        <h4 class="font-semibold">{{ track.meta?.title }}</h4>
+        <span class="text-slate-700" v-if="track.meta?.artist">
+          {{ track.meta?.artist }}
+        </span>
+      </div>
+      <div class="dynamic-spacer"></div>
+      <div>
+        <span class="text-slate-400">{{ track.subtype }}</span>
+      </div>
+      <div class="ml-auto">
+        <span class="text-slate-400">{{ track.meta?.time }}</span>
+      </div>
+      <div class="ml-auto flex justify-center items-center gap-3">
+        <Icon name="download" filled class="only-show-on-hover text-2xl" />
+        <Icon name="queue" filled class="only-show-on-hover text-2xl" />
+        <Icon name="options" filled class="text-2xl" />
+      </div>
     </div>
   </li>
 </template>
+
+<style>
+@tailwind base;
```

Comment:
Using `@apply` is not recommended by TailwindCSS, as it kinda defeats the purpose of using Tailwind in the first case.

- Try using a drop shadow for the hover styling
- use the `group` and `group-hover:` classes to style children based on the parents hover state

Reply:
Done ✅